app_llm.py

```diff
@@ -0,0 +1,355 @@
+
+    current_max_pixel_length = MAXIMUM_PIXEL_LENGTH
+    while byte_size > MAXIMUM_BYTES:
+        quality_scale = MAXIMUM_BYTES / byte_size
+        current_max_pixel_length = int(quality_scale * current_max_pixel_length)
+        img_to_send.thumbnail((current_max_pixel_length, current_max_pixel_length))
+        img_as_bytes = io.BytesIO()
+        img_to_send.save(img_as_bytes, "jpeg")
+        encoded_img = base64.b64encode(img_as_bytes.getvalue()).decode("utf-8")
+        byte_size = encoded_img.__sizeof__()
+
+    anthropic_client = anthropic.Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))
+
+    message = anthropic_client.messages.create(
+        model=ANTHROPIC_MODEL,
+        max_tokens=OUPUT_SIZE_LIMIT,
+        # If set to 0, the model will use log probability 96 to automatically increase the temperature until certain thresholds are hit.
+        temperature=0.1,
+        top_p=0,
+        messages=[
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "image",
+                        "source": {
+                            "type": "base64",
+                            "media_type": "image/jpeg",
+                            "data": encoded_img,
+                        },
+                    },
+                    {
+                        "type": "text",
+                        "text": "Identify all tables and table structure"
+                                "Then hardcore the table structure with columns and rows."
+                                "Extract the number from the tables, fill in the table."
+                                "Make columns use headers field in json file, and data use data field."
+                                "The non_table_data should be key-value pair "
+                                "Respond directly in JSON format without any introduction, explanation, or additional text."
+                    },
+                ],
+            }
+        ]
+    )
+    return json.loads(message.content[0].text)
+
+
+def correct_image_orientation(image_path):
+    """
+    Corrects the orientation of an image based on its EXIF data.
+
+    Parameters:
+    image_path (str): The path to the image file.
+
+    Returns:
+    PIL.Image.Image: The image with corrected orientation.
+    """
+    image = Image.open(image_path)
+    orientation = None
+    try:
+        for orientation in ExifTags.TAGS.keys():
+            if ExifTags.TAGS[orientation] == 'Orientation':
+                break
+        exif = dict(image._getexif().items())
+        if exif.get(orientation) == 3:
+            image = image.rotate(180, expand=True)
+        elif exif.get(orientation) == 6:
+            image = image.rotate(270, expand=True)
+        elif exif.get(orientation) == 8:
+            image = image.rotate(90, expand=True)
+    except (AttributeError, KeyError, IndexError):
+        pass
+    return image
```

Comment:
I think that long-term these two functions should go with the backend code, since these are the type of things that would be put in a serverless function and/or evaluation script. For now leaving them here is fine, but we should keep that in mind for refactoring.

Reply:
Yeah, definitely. I move these to ocr_functions in my new branch.